src/audio/dai-zephyr.c

```diff
@@ -268,16 +268,55 @@ dai_dma_cb(struct dai_data *dd, struct comp_dev *dev, uint32_t bytes,
 	}
 
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
+#if CONFIG_IPC_MAJOR_4
+		struct list_item *sink_list;
+		/*
+		 * copy from local buffer to all sinks that are not gateway buffers
+		 * using the right PCM converter function.
+		 */
+		list_for_item(sink_list, &dev->bsink_list) {
+			struct comp_dev *sink_dev;
+			struct comp_buffer *sink;
+			int j;
+
+			sink = container_of(sink_list, struct comp_buffer, source_list);
+
+			if (sink == dd->dma_buffer)
```

Comment:
dma_buffer is not on the list of sinks, so this condition is always false. However, maybe it is OK to leave it as is or replace with assert() "just in case" someone add dma_buffer to sink list in some future refactoring.